### PR TITLE
Fix #531, Consider an extension of annotation()

### LINKED ISSUE #531: Consider an extension of annotation()

### PR DESCRIPTION
**Pull request by sunghee-hwang**
***
<a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/pull/536.html" title="Last updated on Jun 27, 2023, 5:52 PM UTC (576d538)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/iamf/536/3318721...576d538.html" title="Last updated on Jun 27, 2023, 5:52 PM UTC (576d538)">Diff</a>

### CODE REVIEW

**Review by cconcolato — COMMENTED** [review]
What is the use case?

Also `string_size` is not defined. Does it include the terminating NULL byte?

**Comment by sunghee-hwang** [comment]
> What is the use case?
> 
> Also `string_size` is not defined. Does it include the terminating NULL byte?

We have a postponed issue, https://github.com/AOMediaCodec/iamf/issues/290

File: index.bs [hunk 1]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
 <b>Syntax</b>
 ```
 class mix_presentation_annotations() {
+  leb128() mix_presentation_annotation_size;
   string mix_presentation_friendly_label;
+  unsigned int (8*(mix_presentation_annotation_size - string_size)) mix_presentation_annotation_bytes;
```
Comment (sunghee-hwang) [inline index.bs:1083]:
```suggestion
  leb128() mix_presentation_annotation_size;
  unsigned int (8*mix_presentation_annotation_size) mix_presentation_annotation_bytes;
```

File: index.bs [hunk 2]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
+<dfn noexport>mix_presentation_annotation_bytes</dfn> indicates the byte representations of syntaxes of this mix_presentation_annotation() immediately following [=mix_presentation_friendly_label=] field.
 
 ### Mix Presentation Element Annotations Syntax and Semantics ### {#obu-mixpresentation-elementannotation}
 
 <b>Syntax</b>
 ```
 class mix_presentation_element_annotations() {
+  leb128() mix_presentation_element_annotation_size;
   string audio_element_friendly_label;
+  unsigned int (8*(mix_presentation_element_annotation_size - string_size)) mix_presentation_element_annotation_bytes;
```
Comment (sunghee-hwang) [inline index.bs:1102]:
```suggestion
  leb128() mix_presentation_element_annotation_size;
  unsigned int (8*mix_presentation_element_annotation_size) mix_presentation_element_annotation_bytes;
```

File: index.bs [hunk 3]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
 class mix_presentation_annotations() {
+  leb128() mix_presentation_annotation_size;
   string mix_presentation_friendly_label;
+  unsigned int (8*(mix_presentation_annotation_size - string_size)) mix_presentation_annotation_bytes;
 }
 ```
 
 <b>Semantics</b>
 
+<dfn noexport>mix_presentation_annotation_size</dfn> indicates the size in bytes of this mix_presentation_annotation() immediately following this field.
```
Comment (sunghee-hwang) [inline index.bs:1089]:
```suggestion
```

File: index.bs [hunk 4]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
 <b>Syntax</b>
 ```
 class mix_presentation_annotations() {
+  leb128() mix_presentation_annotation_size;
```
Comment (sunghee-hwang) [inline index.bs:1081]:
```suggestion
```

File: index.bs [hunk 5]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
 }
 ```
 
 <b>Semantics</b>
 
+<dfn noexport>mix_presentation_annotation_size</dfn> indicates the size in bytes of this mix_presentation_annotation() immediately following this field.
+
 <dfn noexport>mix_presentation_friendly_label</dfn> specifies a human-friendly label to describe this [=Mix Presentation=].
 
+<dfn noexport>mix_presentation_annotation_bytes</dfn> indicates the byte representations of syntaxes of this mix_presentation_annotation() immediately following [=mix_presentation_friendly_label=] field.
```
Comment (sunghee-hwang) [inline index.bs:1093]:
```suggestion
<dfn noexport>mix_presentation_annotation_size</dfn> indicates the size in bytes of [=mix_presentation_annotation_bytes=].

<dfn noexport>mix_presentation_annotation_bytes</dfn> represents reserved bytes for future use when new syntaxes are defined. Parsers compliant to this version of the specification SHALL ignore these bytes.
```

File: index.bs [hunk 6]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
 
 ### Mix Presentation Element Annotations Syntax and Semantics ### {#obu-mixpresentation-elementannotation}
 
 <b>Syntax</b>
 ```
 class mix_presentation_element_annotations() {
+  leb128() mix_presentation_element_annotation_size;
```
Comment (sunghee-hwang) [inline index.bs:1100]:
```suggestion
```

File: index.bs [hunk 7]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
+  leb128() mix_presentation_annotation_size;
   string mix_presentation_friendly_label;
+  unsigned int (8*(mix_presentation_annotation_size - string_size)) mix_presentation_annotation_bytes;
 }
 ```
 
 <b>Semantics</b>
 
+<dfn noexport>mix_presentation_annotation_size</dfn> indicates the size in bytes of this mix_presentation_annotation() immediately following this field.
+
```
Comment (sunghee-hwang) [inline index.bs:1090]:
```suggestion
```

File: index.bs [hunk 8]
```diff
@@ -1078,28 +1078,39 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
 }
 ```
 
 <b>Semantics</b>
 
+<dfn noexport>mix_presentation_element_annotation_size</dfn> indicates the size in bytes of this mix_presentation_element_annotation() immediately following this field.
+
 <dfn noexport>audio_element_friendly_label</dfn> specifies a human-friendly label to describe the referenced [=Audio Element=].
 
+<dfn noexport>mix_presentation_element_annotation_bytes</dfn> indicates the byte representations of syntaxes of this mix_presentation_element_annotation() immediately following [=mix_presentation_element_friendly_label=] field.
```
Comment (sunghee-hwang) [inline index.bs:1112]:
```suggestion
<dfn noexport>mix_presentation_element_annotation_size</dfn> indicates the size in bytes of [=mix_presentation_element_annotation_bytes=].

<dfn noexport>mix_presentation_element_annotation_bytes</dfn> represents reserved bytes for future use when new syntaxes are defined. Parsers compliant to this version of the specification SHALL ignore these bytes.
```

**Review by sunghee-hwang — COMMENTED** [review]
To address the comment and more clarification

File: index.bs [hunk 9]
```diff
@@ -1079,27 +1079,40 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
 class mix_presentation_element_annotations() {
   string audio_element_friendly_label;
+  leb128() mix_presentation_element_annotation_size;
+  unsigned int (8*mix_presentation_element_annotation_size) mix_presentation_element_annotation_bytes;
 }
 ```
 
 <b>Semantics</b>
 
+<dfn noexport>mix_presentation_element_annotation_size</dfn> indicates the size in bytes of this mix_presentation_element_annotation() immediately following this field.
```
Comment (sunghee-hwang) [inline index.bs:1108]:
```suggestion
```

File: index.bs [hunk 10]
```diff
@@ -1079,27 +1079,40 @@ Each sub-mix SHALL include [=loudness_layout=] to identify [=Loudspeaker configu
...
+
```
Comment (sunghee-hwang) [inline index.bs:1109]:
```suggestion
```

**Review by sunghee-hwang — COMMENTED** [review]
Deleted the duplicated lines

**Comment by tdaede** [comment]
We chose not implement this as-is as the extension is per-language. We can either define a non-per-language extension point or use the data at the end of the OBU for future extension.

**Pull request closed without merging**